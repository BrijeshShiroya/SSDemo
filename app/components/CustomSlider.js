import PropTypes from 'prop-types';
import React from 'react';
import { View } from 'react-native';
import Slider from 'react-native-slider';
import { Greets } from '../components';
import { Colors } from '../theme';
import styles from './styles/CustomSliderStyles';

const CustomSlider = ({
  title,
  sliderValue,
  onSliderValueChange,
  thumbIcon
}) => {
  return (
    <View>
      <View style={styles.titleContainer}>
        <Greets title={title} style={styles.greet} />
      </View>
      <Slider
        animateTransitions
        value={sliderValue}
        style={styles.sliderContainer}
        thumbImage={thumbIcon}
        thumbStyle={styles.thumb}
        trackStyle={styles.track}
        minimumValue={0}
        maximumValue={1}
        thumbTintColor={Colors.btnOrange}
        minimumTrackTintColor={Colors.txtPlaceholderColor}
        maximumTrackTintColor={Colors.transparent}
        onValueChange={(value) => onSliderValueChange(value)}
      />
    </View>
  );
};

CustomSlider.propTypes = {
  title: PropTypes.string,
  sliderValue: PropTypes.number,
  onSliderValueChange: PropTypes.func,
  thumbIcon: PropTypes.object
};

export default CustomSlider;
