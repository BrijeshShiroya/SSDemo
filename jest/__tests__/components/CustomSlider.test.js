import 'react-native';
import React from 'react';
import CustomSlider from '../../../app/components/CustomSlider';
import { shallow } from 'enzyme';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

const props = {
  onSliderValueChange: jest.fn()
};

const slider = <CustomSlider {...props} />;

it('CustomSlider renders correctly', () => {
  const customSliderComponent = renderer.create(slider).toJSON();
  expect(customSliderComponent).toMatchSnapshot();
});

it('Change Slider value', () => {
  const onSliderValueChange = jest.spyOn(props, 'onSliderValueChange');
  const econ = shallow(
    <CustomSlider sliderValue={0} onSliderValueChange={onSliderValueChange} />
  ).dive();
  econ.instance().props.children[1].props.onValueChange(5);
  expect(onSliderValueChange).toHaveBeenCalled();
});
