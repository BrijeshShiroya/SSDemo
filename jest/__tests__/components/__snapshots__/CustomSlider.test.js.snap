// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSlider renders correctly 1`] = `
<View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "bottom": 0,
        "justifyContent": "center",
        "position": "absolute",
        "top": 16.428571428571427,
        "width": "90%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "white",
            "fontFamily": "AvenirNext-Bold",
            "fontSize": 34,
            "textAlign": "center",
          },
          Object {
            "fontSize": 32,
          },
        ]
      }
    />
  </View>
  <View
    animateTransitions={true}
    animationType="timing"
    onLayout={[Function]}
    onValueChange={[Function]}
    step={0}
    style={
      Array [
        Object {
          "height": 40,
          "justifyContent": "center",
        },
        Object {
          "alignSelf": "center",
          "borderRadius": 4,
          "marginTop": 19.714285714285715,
          "width": "90%",
        },
      ]
    }
    thumbTouchSize={
      Object {
        "height": 40,
        "width": 40,
      }
    }
    value={0}
  >
    <View
      onLayout={[Function]}
      renderToHardwareTextureAndroid={true}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          Object {
            "borderRadius": 2,
            "height": 4,
          },
          Object {
            "borderColor": "rgb(135, 135, 137)",
            "borderWidth": 1,
            "height": 78.85714285714286,
          },
        ]
      }
    />
    <View
      renderToHardwareTextureAndroid={true}
      style={
        Object {
          "backgroundColor": "rgb(247, 247, 247)",
          "borderColor": "rgb(135, 135, 137)",
          "borderRadius": 2,
          "borderWidth": 1,
          "height": 78.85714285714286,
          "opacity": 0,
          "position": "absolute",
          "width": 0,
        }
      }
    />
    <View
      onLayout={[Function]}
      renderToHardwareTextureAndroid={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgb(248, 156, 33)",
          "borderRadius": 4,
          "height": 78.85714285714286,
          "justifyContent": "center",
          "opacity": 0,
          "position": "absolute",
          "transform": Array [
            Object {
              "translateX": 0,
            },
            Object {
              "translateY": 0,
            },
          ],
          "width": 78.85714285714286,
        }
      }
    />
    <View
      onMoveShouldSetResponder={[Function]}
      onMoveShouldSetResponderCapture={[Function]}
      onResponderEnd={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderReject={[Function]}
      onResponderRelease={[Function]}
      onResponderStart={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onStartShouldSetResponderCapture={[Function]}
      renderToHardwareTextureAndroid={true}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {},
        ]
      }
    />
  </View>
</View>
`;
